### PR TITLE
Documenting how to list 3rd party dependencies

### PR DESCRIPTION
**Pull request by gregkalapos**
Documenting the outcome of the license discussion from the [MondoDb PR](https://github.com/elastic/apm-agent-dotnet/pull/1215).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -156,7 +156,19 @@ In case of existing dependencies we need to depend on the oldest possible versio
 
 ### Coding guidelines
 
-The repository contains a [.editorconfig file](.editorconfig) which is automatically picked up by Visual Studio and JetBrains Rider. Additionally, we also have a [.DotSettings file](ElasticApmAgent.sln.DotSettings). If you auto-format your code in Visual Studio or Rider, you will automatically confirm to our code styling. If you want to know more details about the specific conventions we use, feel free to look into the [.editorconfig file](.editorconfig). 
+The repository contains a [.editorconfig file](.editorconfig) which is automatically picked up by Visual Studio and JetBrains Rider. Additionally, we also have a [.DotSettings file](ElasticApmAgent.sln.DotSettings). If you auto-format your code in Visual Studio or Rider, you will automatically confirm to our code styling. If you want to know more details about the specific conventions we use, feel free to look into the [.editorconfig file](.editorconfig).
+
+### 3rd party dependencies
```
Comment (russcam) [inline CONTRIBUTING.md:161]:
```suggestion
### Third party dependencies
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -156,7 +156,19 @@ In case of existing dependencies we need to depend on the oldest possible versio
...
+
+In case an agent project depends on a NuGet package, the dependency does not have to be listed, since it's not directly shipped with the agent. 
+
+However if source code from a 3rd party dependency is added to the agent, the following steps must be taken:
```
Comment (russcam) [inline CONTRIBUTING.md:165]:
```suggestion
However if source code from a third party dependency is added to the agent, the following steps must be taken:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -156,7 +156,19 @@ In case of existing dependencies we need to depend on the oldest possible versio
...
+1. Make sure the license of the 3rd party dependency allows us to take the code.
```
Comment (russcam) [inline CONTRIBUTING.md:166]:
```suggestion
1. Make sure the license of the third party dependency is one that is permitted to be included in the agent, based on Elastic's Open Source Policy
```

**Review by russcam — COMMENTED** [review]
Left some suggestions

**Review by russcam — APPROVED** [review]

**Review by gtback — COMMENTED** [review]
Sorry I didn't have the chance to review this before it was merged. This LGTM!

**Pull request merged**